site/packages/aa-core/smart-account-client/middleware/index.md

```diff
@@ -57,9 +59,19 @@ This is a no-op middleware, but you can include any custom step during the user
 
 If you are simulating a user operation using [`simulateUserOperation`](/packages/aa-alchemy/smart-account-client/actions/simulateUserOperation.md), `SmartAccountClient` will include `userOperationSimulator` middleware during the pipeline run to simulate the user operation instead of sending it to the network to be mined.
 
-## Use custom middleware using `ClientMiddlewareConfig` in `SmartAccountClientConfig`
+During the `SmartAccountClient` middleware pipeline run, each middleware is applied to compute certain fields of the [`UserOperationStruct`](/resources/types#useroperationstruct) in the following sequential order:
+
+1. `dummyPaymasterDataMiddleware` -- populates a dummy paymaster data to use in estimation (default: `0x` for no paymater)
+2. `feeEstimator` -- sets maxfeePerGas and maxPriorityFeePerGas
+3. `gasEstimator` -- calls eth_estimateUserOperationGas
+4. `customMiddleware` -- default no-op middleware, available for any custom operation to be done in addition and override any of the results returned by previous middlewares
+5. `paymasterMiddleware` -- used to set paymaster fields of a user operation. (default: `0x` for no paymater)
+6. `userOperationSimulator` -- used for the simulation of a user operation instead of sending it to the network to be mined.
```

Comment:
I'd be interested in a little more detail here. Does this only get used in the case where simulation is happening so it's a conditional part of the pipeline, or does it always run? Is it expected to update the user op somehow or is it returning other information?

Reply:
Added links to where they can learn more (`alchemyUserOperationSimulator`)